lib/theme_check/shopify_liquid/scope_index.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'pathname'
+
+require_relative 'scope_index/base_entry'
+require_relative 'scope_index/filter_entry'
+require_relative 'scope_index/object_entry'
+require_relative 'scope_index/parameter_entry'
+require_relative 'scope_index/property_entry'
+require_relative 'scope_index/return_type_entry'
+require_relative 'scope_index/tag_entry'
+
+module ThemeCheck
+  module ShopifyLiquid
+    class ScopeIndex
```

Comment:
> When I started thinking about things those elements have in common, and thought - well, they share the scope of a Liquid template (still, that's not entirely accurate for filters).

i see, that makes sense! i first thought of variable scope when i saw `ScopeIndex` but not sure why i didn't associate that with liquid objects 😅 

> I personally believe the prefix Data is a bit general, so I'd avoid that one. Schema might mean a different concept in the theme development, so I'd avoid that one as well.

yeeeeh, good call out 🙏 

> Some other names we might adopt:
> LiquidSourceIndex
> TemplateSourceIndex
> TemplateScopeIndex
> GlobalScopeIndex

i think these are all good! im fine with any of them, i like that it's adding extra specificity so `Scope` can't be confused. `LiquidSourceIndex` might be redundant namespace wise? eg `ThemeCheck:: ShopifyLiquid::LiquidSourceIndex`

> Despite the name, I believe it will be valuable to add a header in this class describing its purpose to ensure we're keeping it clear for future readers 😅

for sure agreed!